src/v/rpc/netbuf.cc

```diff
@@ -32,11 +32,6 @@ ss::future<ss::scattered_message<char>> netbuf::as_scattered() && {
     iobuf out_buf = std::move(_out);
     auto hdr = _hdr;
 
-    if (hdr.correlation_id == 0 || hdr.meta == 0) {
-        throw std::runtime_error(
-          "cannot compose scattered view with incomplete header. missing "
-          "correlation_id or remote method id");
-    }
```

Comment:
>(and it is unlikely that there still remain entries with
these values in the _correlations map)

in the unlikely event this occurred, wouldn't this pair a response with the incorrect request?

Reply:
qq @ztlpn

Reply:
We have a check for this [here](https://github.com/redpanda-data/redpanda/blob/dev/src/v/rpc/transport.cc#L137)

Reply:
ah yes of course i forgot. thanks!